### ISSUE TITLE
missing layers (compared to DGL)

### ISSUE THREAD

**Issue by CarloLucibello** [CarloLucibello/GraphNeuralNetworks.jl#41]
Checklist of layers we miss compared to [Deep Graph Library](https://docs.dgl.ai/api/python/nn.html).
PRs are welcome!

### Conv Layers

- [x]  GraphConv     (called `GCNConv` here)
- [ ] EdgeWeightNorm
- [ ] RelGraphConv
- [ ] TAGConv
- [x] GATConv
- [x] EdgeConv
- [ ] SAGEConv
- [ ]SGConv
- [ ] APPNPConv
- [x] GINConv
- [ x] GatedGraphConv
- [x] GMMConv
- [ ] ChebConv
- [ ] AGNNConv
- [x] NNConv
- [ ] AtomicConv
- [ ] CFConv
- [ ] DotGatConv
- [ ] TWIRLSConv
- [ ] TWIRLSUnfoldingAndAttention
- [ ] GCN2Conv

### Dense Conv Layers

- [ ] DenseGraphConv
- [ ] DenseSAGEConv
- [ ] DenseChebConv

### Global Pooling Layers

- [x] SumPooling (`GlobalPooling(+)` here)
- [x] AvgPooling (`GlobalPooling(mean)` here)
- [x] MaxPooling (`GlobalPooling(max)` here)
- [ ] SortPooling
- [ ] WeightAndSum
- [ ] GlobalAttentionPooling
- [ ] Set2Set
- [ ] SetTransformerEncoder
- [ ] SetTransformerDecoder

**Comment by oysteinsolheim** [comment]
Would it be possible to also have a list of wished-for-algorithms, included or not in DGL? :-) For example I'd love to see edge-featured-based algorithms in general, and maybe more specifically the "Exploiting Edge Features in Graph Neural Networks" from .https://arxiv.org/abs/1809.02709

**Comment by CarloLucibello** [comment]
Of course! You can file a separate issue for each feature request and I'll try to get to them as soon as I can spare some time if no one beats me to it. For instance, "Exploiting Edge Features in Graph Neural Networks" seems to be cited enough that it is worth having, so you are welcome to open a new issue.

**Comment by oysteinsolheim** [comment]
Perfect! :-)
